prow/plugins/label/label.go

```diff
@@ -0,0 +1,133 @@
+	sigMatcher       = regexp.MustCompile(`(?m)@sig-([\w-]*)-(?:misc|test-failures|bugs|feature-requests|proposals|pr-reviews|api-reviews)`)
+	nonExistentLabel = "These labels do not exist in this repository: `%v`"
+	labelFailed      = "The following labels: %v could not be added to this issue"
+)
+
+func init() {
+	plugins.RegisterIssueCommentHandler(pluginName, handleIssueComment)
+}
+
+type githubClient interface {
+	CreateComment(owner, repo string, number int, comment string) error
+	IsMember(org, user string) (bool, error)
+	AddLabel(owner, repo string, number int, label string) error
+	GetLabels(owner, repo string) ([]string, error)
+	BotName() string
+}
+
+func handleIssueComment(pc plugins.PluginClient, ic github.IssueCommentEvent) error {
+	return handle(pc.GitHubClient, pc.Logger, ic)
+}
+
+func handle(gc githubClient, log *logrus.Entry, ic github.IssueCommentEvent) error {
+	commenter := ic.Comment.User.Login
+	owner := ic.Repo.Owner.Name
+	repo := ic.Repo.Name
+	number := ic.Issue.Number
+
+	// only parse newly created comments and if non bot author
+	if commenter == gc.BotName() || ic.Action != "created" {
+		return nil
+	}
+
+	labelMatches := labelRegex.FindAllStringSubmatch(ic.Comment.Body, -1)
+	sigMatches := sigMatcher.FindAllStringSubmatch(ic.Comment.Body, -1)
+	if len(labelMatches) == 0 && len(sigMatches) == 0 {
+		return nil
+	}
+
+	member, err := gc.IsMember(owner, commenter)
+	if err != nil {
+		return fmt.Errorf("IsMember failed: %v", err)
+	}
+	if !member {
+		return gc.CreateComment(owner, repo, number, plugins.FormatResponse(ic.Comment, "Only kubernetes org members can add labels."))
+	}
+
+	labels, err := gc.GetLabels(owner, repo)
+	if err != nil {
+		return err
+	}
+	existingLabels := sets.NewString(labels...)
+	var failures, nonexistent []string
+
+	for _, match := range labelMatches {
+		for _, newLabel := range strings.Split(match[0], " ")[1:] {
+			newLabel = match[1] + "/" + strings.TrimSpace(newLabel)
+			if ic.Issue.HasLabel(newLabel) {
+				continue
+			}
+			if !existingLabels.Has(newLabel) {
+				nonexistent = append(nonexistent, newLabel)
+				continue
+			}
+			if err := gc.AddLabel(owner, repo, number, newLabel); err != nil {
+				failures = append(failures, newLabel)
+			}
+		}
+	}
+
+	for _, sig_match := range sigMatches {
+		sigLabel := "sig" + "/" + strings.TrimSpace(sig_match[1])
+		if ic.Issue.HasLabel(sigLabel) {
+			continue
+		}
+		if !existingLabels.Has(sigLabel) {
+			nonexistent = append(nonexistent, sigLabel)
+			continue
+		}
+		if err := gc.AddLabel(owner, repo, number, sigLabel); err != nil {
+			failures = append(failures, sigLabel)
+		}
+
+	}
+
+	if len(nonexistent) > 0 {
+		if err := gc.CreateComment(owner, repo, number, fmt.Sprintf(nonExistentLabel, strings.Join(nonexistent, ", "))); err != nil {
+			log.Errorf("Could not create comment: %v.  Error = %s \n", fmt.Sprintf(nonExistentLabel, strings.Join(nonexistent, ", ")), err.Error())
+		}
+	}
+
+	if len(failures) > 0 {
```

Comment:
If we get an error adding a label and then post the comment successfully then no error line will show up in prow's logs. We should only get an error adding labels in exceptional cases, such as when GitHub is down. I think in this case it's reasonable to just log the error and not leave a comment.